controller/helpers.go

```diff
@@ -47,6 +48,18 @@ func setLogging() error {
 	return nil
 }
 
+// check to see if a unit update only updates a unit level change
+// logic is: if the two units have idential configs _and_ different log levels, then true
+func onlyLogLevelUpdated(newUnit, currentUnit map[string]interface{}, newLog client.UnitLogLevel) bool {
+	if newUnit == nil || currentUnit == nil {
+		return false
+	}
+	if reflect.DeepEqual(newUnit, currentUnit) && config.ZapFromUnitLogLevel(newLog) != logp.GetLevel() {
```

Comment:
Can the `interface{}` portion of the unit variables be a protobuf message? If it can this isn't correct and won't evaluate deterministically. You need to use `proto.Equal` for that comparison to behave properly.

Reply:
Nope, this is just the interface from the `source` field, so it should just me the raw config, without any protobuf fields.